demos/multi_material/thermochemical_buoyancy/thermochemical_buoyancy.py

```diff
@@ -1,101 +1,102 @@
+    point_1 = gmsh.model.geo.addPoint(0.0, 0.0, 0.0, mesh_hor_res)
+    point_2 = gmsh.model.geo.addPoint(domain_dims[0], 0.0, 0.0, mesh_hor_res)
 
-line_1 = gmsh.model.geo.addLine(point_1, point_2)
+    line_1 = gmsh.model.geo.addLine(point_1, point_2)
 
-gmsh.model.geo.extrude(
-    [(1, line_1)], 0, ly / 5, 0, numElements=[40], recombine=True
-)  # Vertical resolution: 5e-3
+    gmsh.model.geo.extrude(
+        [(1, line_1)], 0.0, 0.1, 0.0, numElements=[20], recombine=True
+    )  # Vertical resolution: 5e-3
 
-gmsh.model.geo.extrude(
-    [(1, line_1 + 1)], 0, ly - ly / 5 - ly / 20, 0, numElements=[15], recombine=True
-)  # Vertical resolution: 5e-2
+    gmsh.model.geo.extrude(
+        [(1, line_1 + 1)], 0.0, 0.8, 0.0, numElements=[16], recombine=True
+    )  # Vertical resolution: 5e-2
 
-gmsh.model.geo.extrude(
-    [(1, line_1 + 5)], 0, ly / 20, 0, numElements=[10], recombine=True
-)  # Vertical resolution: 5e-3
+    gmsh.model.geo.extrude(
+        [(1, line_1 + 5)], 0.0, 0.1, 0.0, numElements=[20], recombine=True
+    )  # Vertical resolution: 5e-3
 
-gmsh.model.geo.synchronize()
+    gmsh.model.geo.synchronize()
 
-gmsh.model.addPhysicalGroup(1, [line_1 + 2, line_1 + 6, line_1 + 10], tag=1)
-gmsh.model.addPhysicalGroup(1, [line_1 + 3, line_1 + 7, line_1 + 11], tag=2)
-gmsh.model.addPhysicalGroup(1, [line_1], tag=3)
-gmsh.model.addPhysicalGroup(1, [line_1 + 9], tag=4)
+    gmsh.model.addPhysicalGroup(1, [line_1 + 2, line_1 + 6, line_1 + 10], tag=1)
+    gmsh.model.addPhysicalGroup(1, [line_1 + 3, line_1 + 7, line_1 + 11], tag=2)
+    gmsh.model.addPhysicalGroup(1, [line_1], tag=3)
+    gmsh.model.addPhysicalGroup(1, [line_1 + 9], tag=4)
 
-gmsh.model.addPhysicalGroup(2, [line_1 + 4, line_1 + 8, line_1 + 12], tag=1)
+    gmsh.model.addPhysicalGroup(2, [line_1 + 4, line_1 + 8, line_1 + 12], tag=1)
 
-gmsh.model.mesh.generate(2)
+    gmsh.model.mesh.generate(2)
 
-gmsh.write(mesh_file)
-gmsh.finalize()
+    gmsh.write(mesh_file)
+    gmsh.finalize()
 # -
 
 # We next set up the mesh and function spaces and specify functions to hold our
 # solutions, as in our previous tutorials.
 
 # +
 mesh = Mesh(mesh_file)  # Load the GMSH mesh using Firedrake
-mesh.cartesian = True
+mesh.cartesian = True  # Tag the mesh as Cartesian to inform other G-ADOPT objects
+boundary = get_boundary_ids(mesh)  # Object holding references to mesh boundary IDs
 
-boundary = get_boundary_ids(mesh)
-
-V = VectorFunctionSpace(mesh, "CG", 2)  # Velocity function space (vector)
-W = FunctionSpace(mesh, "CG", 1)  # Pressure function space (scalar)
+V = VectorFunctionSpace(mesh, "Q", 2)  # Velocity function space (vector)
+W = FunctionSpace(mesh, "Q", 1)  # Pressure function space (scalar)
 Z = MixedFunctionSpace([V, W])  # Stokes function space (mixed)
-Q = FunctionSpace(mesh, "CG", 2)  # Temperature function space (scalar)
+# Temperature function space (scalar, discontinuous, equispaced)
+Q = FunctionSpace(mesh, "DQ", 2, variant="equispaced")
```

Comment:
A comment motivating equispaced and other options would be useful - this will be new to the user.